### ISSUE TITLE
500 ERROR on login

### ISSUE THREAD

**Issue by juracana** [mouneyrac/moodle-auth_googleoauth2#126]
Hi, I already set the google id and secret key but when i try to login i got a 500 error, dont why, please helpme 

![login3](https://cloud.githubusercontent.com/assets/10249635/8150722/2fd152ba-12b2-11e5-8ed8-10445a235497.PNG)
![login4](https://cloud.githubusercontent.com/assets/10249635/8150723/2fd16aca-12b2-11e5-82ae-da23314e0eff.PNG)
![login1](https://cloud.githubusercontent.com/assets/10249635/8150724/2fd2208c-12b2-11e5-8b2b-abf889d70a5c.PNG)
![login2](https://cloud.githubusercontent.com/assets/10249635/8150725/2fd6c1dc-12b2-11e5-9cd5-3bfd3bb32058.PNG)

**Comment by spichoff** [comment]
I am in the same situation and would love to know if you have this fixed? I upgraded to 2.9 from 2.6 yesterday. I used the same Oauth credentials and then made new ones, and neither are working, but oddly enough my users can still access their google drive though the same creditials I had made for them back in 2.6. 

Estoy con la misma situación y me gustaría oír si Ud. ha encontrado una resolución. Yo hice un upgrade a 2.9 de 2.6 ayer. Usé los mismos credenciales del Google API, y después creé nuevos credenciales pero de todos modos ningunos trabajaron (todavía con el 500 ERROR). Pero lo que encuentro raro es que mis usuarios del sistema tienen acceso a sus cuentas de Google Drive con lo credenciales que creé durante 2.6

**Comment by spichoff** [comment]
I found a solution. I was still registered as 2.6 on the moodle.org website. So as I was getting the plug in version for 2.6. When I changed my version number on the website, and reinstalled the Oauth2 Plugin then it started working.

Encontré la solución. En el sitio de Moodle.org yo tenia versión 2.6 puesto. Así todo los plugins era de 2.6 cuando debe haber ido de 2.9. Lo cambie mi versión en moodle.org a 2.9 be instalé el plugin que corresponde despues del upgrade. Ahora todo funciona como debe.

**Comment by juracana** [comment]
thank you for your answer but i have another 403 error while trying to register my moodle. :(

**Comment by mouneyrac** [comment]
Spichoff: thanks for explaining the solution, I really appreciate it! It avoid me to do some questions/researches :) This is likely to be a Moodle.org problem if Moodle.org doesn't download the correct version matching your Moodle version. So Closing this issue (feel free to comment or open a new issue if you think I am wrong)

juracana: thanks for reporting the issue (it is obvious now that we have the solution that it is not plugin related but we didn't know before). About your second issue, registering your Moodle site on Moodle.org is not related to googleoauth2 authentication. Also it is better to create a new issue when an issue is different. Someone did that recently on another issue and it was quite confusing as someone else came back on the issue to talk about the initial problem. 

I understand this comment may feel a bit un-nice but it is important to know that, as the creator/maintainer, I don't use this plugin, the average money per hours that this plugin generates is much closer from 0 than from an average australian salary rate, I am a professional Moodle developer, I have a family, bla bla bla - very common open source plugin rant :) I think it's important to mention my situation with this plugin time to time as I do understand that people do not know it and may believe as it is a popular plugin and a security prone one, it is well professional supported. It is not the case anymore (I tried to offer such service but no-one got it :)), even though I believe it is well supported for a third party plugin. I am just maintaining for the fun :)

Thanks guys :)
